src/buildercore/core.py

```diff
@@ -85,6 +88,24 @@ def _set_raw_subscription_attribute(sns_connection, subscription_arn):
 
 sns.connection.SNSConnection.set_raw_subscription_attribute = _set_raw_subscription_attribute
 
+def all_sns_subscriptions(region, stackname=None):
+    """returns all subscriptions to all sns topics.
+    optionally filtered by subscription endpoints matching given stack"""
+    conn = boto_sns_conn(region)
+    response = conn.get_all_subscriptions() # TODO: needs pagination!
+    subs_list = response['ListSubscriptionsResponse']['ListSubscriptionsResult']['Subscriptions']
+    if stackname:
+        ''' looks like:
+        {u'Endpoint': u'arn:aws:sqs:us-east-1:512686554592:observer--substest1',
+         u'Owner': u'512686554592',
+         u'Protocol': u'sqs',
+         u'SubscriptionArn': u'arn:aws:sns:us-east-1:512686554592:bus-articles--substest1:f44c42db-81c0-4504-b3de-51b0fb1099ff',
+         u'TopicArn': u'arn:aws:sns:us-east-1:512686554592:bus-articles--substest1'}'''
+        subs_list = filter(lambda row: row['Endpoint'].endswith(stackname), subs_list)
```

Comment:
`Endpoint` seems a strange term for this, but from this response it really looks like the only place that can be used for filtering

Reply:
tell me about it, I spent far too long trying to figure out just what it meant and how I could get access to what I need through which api.